pkg/src/lambdas/functions/geolocation.post.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable */
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'
+import LambdaRequest from '../types/lamdaRequest'
+// import LambdaResponse from '../types/lambdaResponse'
+
+import geocodeSchema from '../schemas/geocode-schema'
+import MapBox from '../../geolocation/mapbox/mapbox.geo'
+
+/**
+ * fetchs lat and long coordinates by checking the given address
+ * with the given provider
+ * @param e the lambda's request object
+ */
+export const geolocationPost = async (e: APIGatewayProxyEvent, result: APIGatewayProxyResult) =>
+{
+	try
+	{
+		const payload = LambdaRequest.validate(e, geocodeSchema)
+		const mapBox = new MapBox(payload[`address`] as string)
+		await mapBox.getCoordinates()
```

Comment:
Why not return the coordinate information from the `getCoordinates` function instead of making the MapBox service stateful?

Reply:
Because each provider can have their own response format. In the mapbox case we define it's response format with the file 'mapbox.response.ts'.

So, the get coordinates is the responsible for fetching the data (the default behaviour), and the mapbox instance itself is the one who I set to be responsible for parsing the response.

I think that we can have less coupling like this, and each provider is free to handle data as they want.